docs/rst/tutorials/cloud/debug_ros2_router/debug_ros2_router.rst

```diff
@@ -0,0 +1,248 @@
+
+
+Advance debugging
+-----------------
+
+The ROS 2 Router can be built with the built-in debug traces so that the internal behavior of the software can be fully monitored.
+To do so, the user/developer should recompile the ROS 2 Router package that comes with Vulcanexus by adding the required compilation options that enable the debug operation mode.
+
+This provides some additional options to filter the debug information shown by the ROS 2 Router. We will explain this in detail below.
+
+.. note::
+
+    Do not mistake the debug mode of ROS 2 Router with the ``Debug`` compilation of C++ code, since ROS 2 Router is still compiled in ``Release`` mode but with debug traces.
+
+Building ROS 2 Router from sources
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Building the ROS 2 Router from sources in Vulcanexus is straightforward as Vulcanexus ships with all the necessary tools for this task, so there is no prerequisite other than the one already shown in this tutorial.
+
+The steps to follow are described below:
+
+1.  Load the Vulcanexus environment.
+
+    -   If you are running Vulcanexus from the distributed Docker image, simply run the Vulcanexus Docker container.
+
+        .. code-block:: bash
+
+            docker run -it ubuntu-vulcanexus:humble-desktop
+
+    -   If you have installed Vulcanexus on your local machine, load the environment with the following command.
+
+        .. code-block:: bash
+
+            source /opt/vulcanexus/humble/setup.bash
+
+1.  Create the development workspace, download the ROS 2 Router from GitHub, and build it by executing the following commands:
+
+    .. code-block:: bash
+
+        mkdir -p ddsrouter_ws/src
+        cd ddsrouter_ws
+        git clone https://github.com/eProsima/DDS-Router src/ddsrouter
+        colcon build --cmake-args -DLOG_INFO=ON
+
+Running ROS 2 Router
+^^^^^^^^^^^^^^^^^^^^
+
+Update the environment setup to use the built ROS 2 Router instead of the one delivered in Vulcanexus.
+
+.. code-block:: bash
+
+    source ddsrouter_ws/install/setup.bash
+
+Monitor ROS 2 Router internal operation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ROS 2 Router offers several input arguments to configure the information displayed when it is built with the internal debug traces.
+These are:
+
+-   ``--log-verbosity <info|warning|error>``: set the verbosity level so only log messages with equal or higher importance level are shown.
+-   ``--log-filer <regex>``: set a regex string as filter.
+    Only log messages with a category that matches this regex will be printed (``ERROR`` messages will be always shown unless ``log-verbosity`` argument is set to ``ERROR``).
+-   ``--debug``: set ``log-verbosity`` to ``info`` and ``log-filter`` to ``DDSROUTER``.
+
+Thus, ROS 2 Router can be run as follows:
+
+.. code-block::
+
+    ddsrouter -c <path/to/file>/echo.yaml --log-verbosity warning --log-filer DDSROUTER_DISCOVERY_DATABASE
```

Comment:
Please, add an example of a trace of a router execution.
e.g. 
`2022-11-14 09:42:36.735 [DDSROUTER_DISCOVERY_DATABASE Info] Creating queue processing thread. -> Function DiscoveryDatabase`